packages/flame/lib/src/effects/controllers/mixins/has_single_child_effect_controller.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:flame/effects.dart';
+import 'package:meta/meta.dart';
+
+/// This mixin must be used with [EffectController]s that wrap a single child
+/// effect controller of type [T].
+mixin HasSingleChildEffectController<T extends EffectController>
```

Comment:
Doesn't all effect controllers wrap a single child, so that this could be directly on the `EffectController` instead of a mixin? 🤔

Reply:
`DurationEffectController`s do not need a child and `SequenceEffectContoller` has a list of children. So these two controllers won't satisfy the single child condition.